flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1675,15 +1703,12 @@ private boolean isJobManagerConnectionValid(JobID jobId, JobMasterId jobMasterId
 			return CompletableFuture.supplyAsync(() -> {
 				final File file = new File(filePath);
 				if (file.exists()) {
-					final TransientBlobCache transientBlobService = blobCacheService.getTransientBlobService();
-					final TransientBlobKey transientBlobKey;
-					try (FileInputStream fileInputStream = new FileInputStream(file)) {
-						transientBlobKey = transientBlobService.putTransient(fileInputStream);
-					} catch (IOException e) {
+					try {
+						return putTransientBlobStream(new FileInputStream(file), fileTag).get();
+					} catch (Exception e) {
```

Comment:
hi @tillrohrmann, since doesn't make thread dump as stream in your design right now, we can revert it.